xarray/core/dataarray.py

```diff
@@ -3425,6 +3838,53 @@ def broadcast_equals(self: T_DataArray, other: T_DataArray) -> bool:
         --------
         DataArray.equals
         DataArray.identical
+
+        Examples
+        --------
+        >>> a = xr.DataArray([1, 1], dims="X")
+        >>> b = xr.DataArray([[1, 1], [1, 1]], dims=["X", "Y"])
+        >>> c = xr.DataArray([2, 2], dims="Y")
+        >>> a
+        <xarray.DataArray (X: 2)>
+        array([1, 1])
+        Dimensions without coordinates: X
+        >>> b
+        <xarray.DataArray (X: 2, Y: 2)>
+        array([[1, 1],
+               [1, 1]])
+        Dimensions without coordinates: X, Y
+        >>> c
+        <xarray.DataArray (Y: 2)>
+        array([2, 2])
+        Dimensions without coordinates: Y
+
+        >>> a2, b2 = xr.broadcast(a, b)
+        >>> a2
+        <xarray.DataArray (X: 2, Y: 2)>
+        array([[1, 1],
+               [1, 1]])
+        Dimensions without coordinates: X, Y
+        >>> b2
+        <xarray.DataArray (X: 2, Y: 2)>
+        array([[1, 1],
+               [1, 1]])
+        Dimensions without coordinates: X, Y
+        >>> a.broadcast_equals(b)
+        True
```

Comment:
Isn't broadcasting them before using `broadcast_equals` sort of missing the point of the method? I would think it would be clear to have
- define `a`, `b`
- `a.equals(b)` -> `False`
- `a.broadcast_equals(b)` -> `True`

Then perhaps with 
- `a2, b2 = xr.broadcast(a, b)`
- `a2.equals(b2)` -> `True`
also to help explain. I think this example could do with some motivating lines of text talking about the difference between `.equals` and `.broadcast_equals`.

Reply:
This is a good suggestion would something like this be better?

        >>> a = xr.DataArray([1, 1], dims="X")
        >>> b = xr.DataArray([[1, 1], [1, 1]], dims=["X", "Y"])
        >>> a
        <xarray.DataArray (X: 2)>
        array([1, 1])
        Dimensions without coordinates: X
        >>> b
        <xarray.DataArray (X: 2, Y: 2)>
        array([[1, 1],
               [1, 1]])
        Dimensions without coordinates: X, Y
        
        .equals returns True if two DataArrays have the same values, dimensions, and coordinates. .broadcast_equals returns True if the results of broadcasting two DataArrays against eachother have the same values, dimensions, and coordinates.
        
        >>> a.equals(b)
        False
        >>> a2, b2 = xr.broadcast(a, b)
        >>> a2.equals(b2)
        True
        >>> 
        >>> a.broadcast_equals(b)
        True

Reply:
Yes much better! I would use `[1, 2]` for the values of `a` though, just to reduce the ambiguity a bit.

Reply:
Fixed. Thanks!